### PR TITLE
GHA: add a swift-format build stage for the toolchain

### PR DESCRIPTION
**Pull request by compnerd**
Begin building `swift-format` as part of the toolchain as this is meant to be part of the toolchain distribution on Windows.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @tristanlabelle @mangini @hjyamauchi

File: .github/workflows/swift-toolchain.yml [hunk 1]
```diff
@@ -1915,9 +1918,84 @@ jobs:
...
+
+      - uses: actions/checkout@v3
+        with:
+          repository: apple/swift-installer-scripts
+          ref: ${{ needs.context.outputs.swift_installer_scripts_revision }}
+          path: ${{ github.workspace }}/SourceCache/swift-installer-scripts
+
+      - uses: microsoft/setup-msbuild@v1.3.1
+
+      - run: msbuild ${{ github.workspace }}/SourceCache/swift-installer-scripts/platforms/Windows/swift-format.wixproj -nologo -restore -p:Configuration=Release -p:SWIFT_FORMAT_BUILD=${{ github.workspace }}\.build\release -p:OutputPath=${{ github.workspace }}\artifacts -p:RunWixToolsOutOfProc=true
```
Comment (hjyamauchi) [inline .github/workflows/swift-toolchain.yml:1987]: Could `SWIFT_FORMAT_BUILD` need to be `${{ github.workspace }}/SourceCache/swift-format/.build/release`? I'm still fuzzy on how the concept of the current directory on a github workflow.

Comment (hjyamauchi) [reply .github/workflows/swift-toolchain.yml:1987]: Did this work? I have a similar setup where it complains that the .exe file isn't found by msbuild at the equivalent of `${{ github.workspace }}/SourceCache/swift-format/.build/release`.

Comment (compnerd) [reply .github/workflows/swift-toolchain.yml:1987]: It is still running (https://github.com/thebrowsercompany/swift-build/actions/runs/5146888554)

Comment (compnerd) [reply .github/workflows/swift-toolchain.yml:1987]: An it passes!

**Pull request merged**